lib/rules/no-return-await.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * @fileoverview Disallows unnecessary `return await`
+ * @author Jordan Harband
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow unnecessary `return await`",
+            category: "Best Practices",
+            recommended: false // TODO: set to true
+        },
+        fixable: false,
+        schema: [
+        ]
+    },
+
+    create(context) {
+        return {
+            ArrowFunctionExpression(node) {
+                if (node.async && node.body.type === "AwaitExpression") {
+                    const sourceCode = context.getSourceCode();
+
+                    context.report({
+                        node: sourceCode.getFirstToken(node.body),
+                        loc: node.body.loc,
+                        message: "await in an implicit-return async arrow function is redundant - just return the value directly.",
+                    });
+                }
+            },
+            ReturnStatement(node) {
+                const argument = node.argument;
+
+                if (argument && argument.type === "AwaitExpression") {
+                    const sourceCode = context.getSourceCode();
+                    const loc = argument.loc;
+                    const containingFunction = node.parent.parent;
```

Comment:
Fixed - i wrote a little abstraction to cleanly get the containing function that should handle all cases.